File: src/RehabEasy.Domain/Models/PatientHistorySnapshot.cs
namespace RehabEasy.Domain.Models;

public sealed class PatientHistorySnapshot
{
    public string PatientId { get; init; } = string.Empty;
    public string? PatientName { get; init; }
    public IReadOnlyList<PatientTestHistoryEntry> Tests { get; init; } = Array.Empty<PatientTestHistoryEntry>();
    public IReadOnlyList<PatientClinicalNoteHistoryEntry> ClinicalNotes { get; init; } =
        Array.Empty<PatientClinicalNoteHistoryEntry>();
    public PatientClinicalNote? CurrentClinicalNote { get; init; }
}

File: java/src/main/java/com/rehabeasy/model/PatientHistorySnapshot.java
package com.rehabeasy.model;

import java.util.List;

public record PatientHistorySnapshot(
        String patientId,
        String patientName,
        List<PatientTestHistoryEntry> tests,
        List<PatientClinicalNoteHistoryEntry> clinicalNotes,
        PatientClinicalNote currentClinicalNote
) {
    public PatientHistorySnapshot {
        patientId = patientId == null ? "" : patientId;
        tests = tests == null ? List.of() : List.copyOf(tests);
        clinicalNotes = clinicalNotes == null ? List.of() : List.copyOf(clinicalNotes);
    }
}
